src/components/documentManager.ts

```diff
@@ -0,0 +1,678 @@
+        }
+    }
+
+    /**
+     * Get url for primary image for a given item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {string | null} url to primary image
+     */
+    private static getPrimaryImageUrl(item: BaseItemDto): string | null {
+        if (item.AlbumPrimaryImageTag)
+            return JellyfinApi.createUrl(
+                `Items/${item.AlbumId}/Images/Primary?tag=${item.AlbumPrimaryImageTag}`
+            );
+        else if (item.ImageTags?.Primary)
+            return JellyfinApi.createUrl(
+                `Items/${item.Id}/Images/Primary?tag=${item.ImageTags.Primary}`
+            );
+        else return null;
+    }
+
+    /**
+     * Get url for logo image for a given item
+     *
+     * @param {BaseItemDto} item to look up
+     * @returns {string | null} url to logo image
+     */
+    private static getLogoUrl(item: BaseItemDto): string | null {
+        if (item.ImageTags?.Logo)
+            return JellyfinApi.createUrl(
+                `Items/${item.Id}/Images/Logo/0?tag=${item.ImageTags.Logo}`
+            );
+        else if (item.ParentLogoItemId && item.ParentLogoImageTag)
+            return JellyfinApi.createUrl(
+                `Items/${item.ParentLogoItemId}/Images/Logo/0?tag=${item.ParentLogoImageTag}`
+            );
+        else return null;
+    }
+
+    /**
+     * This fucntion takes an item and shows details about it
+     * on the details page. This happens when no media is playing,
+     * and the connected client is browsing the library.
+     *
+     * @param {BaseItemDto} item to show information about
+     */
+    public static showItem(item: BaseItemDto): void {
+        // no showItem for cc audio
+        if (getActiveDeviceId() === deviceIds.AUDIO) return;
+
+        // stop cycling backdrops
+        this.clearBackdropInterval();
+
+        this.setAppStatus('details');
+        this.setWaitingBackdrop(item);
+
+        this.setLogo(this.getLogoUrl(item));
+        this.setOverview(item.Overview ?? null);
+        this.setGenres(item?.Genres?.join(' / ') ?? null);
+        this.setDisplayName(item);
+
+        this.setMiscInfo(item);
+
+        const detailRating = document.getElementById('detailRating');
+        if (detailRating) {
+            detailRating.innerHTML = this.getRatingHtml(item);
+        }
+
+        const playedIndicator = document.getElementById('playedIndicator');
+
+        if (playedIndicator) {
+            if (item?.UserData?.Played) {
+                playedIndicator.style.display = 'block';
+                playedIndicator.innerHTML =
+                    '<span class="glyphicon glyphicon-ok"></span>';
+            } else if (item?.UserData?.UnplayedItemCount) {
+                playedIndicator.style.display = 'block';
+                playedIndicator.innerHTML = item.UserData.UnplayedItemCount.toString();
+            } else {
+                playedIndicator.style.display = 'none';
+            }
+        }
```

Comment:
**This one requires changes**: First of all, I think we should add all the relevant HTML tags to the HTML file itself. The ones that won't appear by default (like this one) can have a ``d-none`` class. I would also create another ``d-block`` css class for ``display: block`` property.

Then, I would create a function like ``toggleVisibility`` which takes an HTMLElement as a parameter and toggles the ``d-none`` class. You're already setting ``display: none`` on initialize() and here, so you can reuse that function. You can also make one for the ``d-block``.

This would lead to better readability.

Reply:
we have a .hide class, not sure if it's used